include/rwqueue.h

```diff
@@ -39,14 +39,27 @@
 #include <atomic>
 #include <condition_variable>
 #include <deque>
+
+#ifdef HAVE_MEMORY_RESOURCE
+	#include <memory_resource>
+#endif
+
 #include <mutex>
 #include <optional>
 #include <vector>
 
 template <typename T>
 class RWQueue {
 private:
-	std::deque<T> queue{}; // faster than: vector, queue, and list
+#ifdef HAVE_MEMORY_RESOURCE
+	std::pmr::unsynchronized_pool_resource pool = {};
+
+	// The pooled deque recycles objects from the pool. It only allocates
+	// when it needs to grow beyond its prior peak size.
+	std::pmr::deque<T> queue = {};
+#else
+	std::deque<T> queue = {}; // faster than: vector, queue, and list
```

Comment:
Minor: No end of line comments please unless absolutely necessary (or in case blocks for port descriptions).

Reply:
Will fix that (it was an existing line of code; I only added the PMR chunk).

Reply:
I would just get rid of that comment honestly.

"faster than vector" - Needs benchmark to really prove that.

"faster than queue" - Misleading.  std::queue is only an interface that uses std::deque as the implementation by default.

"faster than list" - Probably true.  Friends don't let friends use linked lists.

Reply:
snipped!